### PR TITLE
Add conversations

### PR DESCRIPTION
**Pull request by epels**
This PR adds support for the Conversations API. See [docs](https://developers.messagebird.com/docs/conversations).

### CODE REVIEW

File: conversation/conversation.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	Status ConversationStatus `json:"status"`
+}
+
+// DefaultListOptions provide a reasonable default for paginated endpoints.
+var DefaultListOptions = &ListOptions{10, 0}
+
+// List gets a collection of Conversations. Pagination can be set in options.
+func List(c *messagebird.Client, options *ListOptions) (*ConversationList, error) {
+	convList := &ConversationList{}
+	if err := request(c, convList, http.MethodGet, path, options); err != nil {
```
Comment (epels) [inline conversation/conversation.go]: Self review: add query parameters to the path. The value for `data` is sent as a JSON encoded body, which doesn't work with GET.

File: conversation/message.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	return message, nil
+}
+
+// ListMessages gets a collection of messages from a conversation. Pagination
+// can be set in the options.
+func ListMessages(c *messagebird.Client, conversationID string, options *ListOptions) (*MessageList, error) {
+	uri := fmt.Sprintf("%s/%s/%s", path, conversationID, messagesPath)
+
+	messageList := &MessageList{}
+	if err := request(c, messageList, http.MethodGet, uri, options); err != nil {
```
Comment (epels) [inline conversation/message.go]: Self review: add query parameters to the path. The value for `data` is sent as a JSON encoded body, which doesn't work with GET.

File: conversation/webhook.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// DeleteWebhook ensures an existing webhook is deleted and no longer
+// triggered. If the error is nil, the deletion was successful.
+func DeleteWebhook(c *messagebird.Client, id string) error {
+	return request(c, nil, http.MethodDelete, webhooksPath+"/"+id, nil)
+}
+
+// ListWebhooks gets a collection of webhooks. Pagination can be set in options.
+func ListWebhooks(c *messagebird.Client, options *ListOptions) (*WebhookList, error) {
+	webhookList := &WebhookList{}
+	if err := request(c, webhookList, http.MethodGet, webhooksPath, options); err != nil {
```
Comment (epels) [inline conversation/webhook.go]: Self review: add query parameters to the path. The value for `data` is sent as a JSON encoded body, which doesn't work with GET.

File: conversation/api.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+type Conversation struct {
+	ID                   string
+	ContactID            string
+	Contact              *Contact
+	LastUsedChannelID    string
+	Channels             []*Channel
+	Messages             *MessagesCount
+	Status               ConversationStatus
+	CreatedDatetime      *time.Time
```
Comment (jamesmccann) [inline conversation/api.go]: CreatedDatetime is always present in response

File: conversation/conversation.go (hunk 1 above)
Comment (epels) [reply conversation/conversation.go]: Resolved in https://github.com/messagebird/go-rest-api/pull/49/commits/430d19fb45c9cf0ee346a0d76a3b611ca78cf2be

File: conversation/message.go (hunk 2 above)
Comment (epels) [reply conversation/message.go]: Resolved in https://github.com/messagebird/go-rest-api/pull/49/commits/430d19fb45c9cf0ee346a0d76a3b611ca78cf2be

File: conversation/webhook.go (hunk 3 above)
Comment (epels) [reply conversation/webhook.go]: Resolved in https://github.com/messagebird/go-rest-api/pull/49/commits/430d19fb45c9cf0ee346a0d76a3b611ca78cf2be

File: conversation/api.go [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+const (
+	MessageStatusDeleted     MessageStatus = "deleted"
+	MessageStatusDelivered   MessageStatus = "delivered"
+	MessageStatusFailed      MessageStatus = "failed"
+	MessageStatusPending     MessageStatus = "pending"
+	MessageStatusRead        MessageStatus = "read"
+	MessageStatusReceived    MessageStatus = "received"
+	MessageStatusSent        MessageStatus = "sent"
+	MessageStatusUnsupported MessageStatus = "unsupported"
+)
```
Comment (jamesmccann) [inline conversation/api.go]: also need "rejected" here

**Pull request merged**